### PR TITLE
WIP provide API to seed a whisper symkey

### LINKED ISSUE #661: Add api call to generate random passphrase
# Problem

As a status-react developper, I would like to be able to use the random passphrase generator of status-go to have a proper random seed for symetric keys.

# Implementation

Add an API call `generate-passphrase` to status-go api. Extract and export it from `geth/account/accounts.go` `CreateAccount` method.

# Acceptance Criteria

I can create a passphrase on status-react side using this API call

# Notes

A function that creates the symkey and returns the passphrase and symkey-id might be even better and avoid a round-trip `generate-sym--key` instead of `generate-passphrase -> generate-sym-key-from-password`

### PR DESCRIPTION
**Pull request by feuGeneA**
Closes #661

### CODE REVIEW

File: geth/api/backend.go [hunk 1]
```diff
@@ -175,6 +176,27 @@ func (b *StatusBackend) ResetChainData() error {
...
+	if err != nil {
+		return "", "", err
+	}
+
+	whisperService, err := m.nodeManager.WhisperService()
+	if err != nil {
+		return "", "", err
+	}
+
+	skid, err := whisperService.AddSymKeyFromPassword(mnemonic)
```
Comment (dshulyak) [inline geth/api/backend.go:192]: this function should not have any side effects on whisper or any other state

File: lib/library.go [hunk 2]
```diff
@@ -106,6 +106,15 @@ func CallRPC(inputJSON *C.char) *C.char {
 	return C.CString(outputJSON)
 }
 
+//SeedWhisperSymKey seeds a random mnemonic passphrase and a Whisper symmetric
+//key seeded with that passphrase as its password, returned via output function
+//parameters.
+//export SeedWhisperSymKey
+func SeedWhisperSymKey(passphrase *C.char, symKeyID *C.char) {
```
Comment (dshulyak) [inline lib/library.go:113]: i don't like the idea of extending library interface with such helpers, if they are needed i would suggest to add another json rpc api (statusdebug). do we need this API for production or only for tests? 

cc @adambabik

**Review by dshulyak — DISMISSED** [review]

**Comment by yenda** [comment]
@JekaMas No storing the key is expected, passphrase is for the other side to be able to generate the symkey

File: geth/api/backend.go (hunk 1 above)
Comment (dshulyak) [reply geth/api/backend.go:192]: just to clarify what i mean, there should be only one channel for making changes to internal state of whisper and this channel is `shh` api

File: lib/library.go (hunk 2 above)
Comment (JekaMas) [reply lib/library.go:113]: This was asked by @yenda and this is for production usage.

Account manager's methods `SelectAccount, ReSelectAccount, Logout` changes the state of a whisper. And all of them are exposed in the library. I think this PR is about exposing one more method from Account manager.

Comment (dshulyak) [reply lib/library.go:113]: oh well, you are right

**Review by dshulyak — COMMENTED** [review]
.

**Comment by ChrisSalisbury** [comment]
For trustless decentralization, any sort of keygen or seedgen should be done on the hardware owned by the entity using said key or seed.

**Comment by feuGeneA** [comment]
@ChrisSalisbury That is what's happening here. This is a C API to interface with the instance of status-go that's running in the same process (and on the same hardware) as status-react.

**Comment by ChrisSalisbury** [comment]
@feuGeneA well, that's good at least. I'll step it back to the comment I left on issue 661 then. Nobody plans on storing it now, but who knows how someone might try to use it in the future (or who might try to hide something that seems innocuous that copies the keys/seeds out somewhere): 

This would create an extremely high motivation to attack this area. If something like this is desired or necessary, the amount of review and debugging it should receive is enormous...I would recommend extending a very large bug bounty on this type of feature for several months before going live with it. (Edit: following best practices to start small and ramp up the bug bounty of course)

**Comment by ChrisSalisbury** [comment]
With examples of things like spectre and meltdown having existed unknown for years, the odds of someone being able to get access to the memory that this is temporarily stored in at some point seem too high to conscionably include this type of feature...

**Comment by feuGeneA** [comment]
Per https://github.com/status-im/status-go/issues/661#issuecomment-366527653

**Pull request closed without merging**